eZ/Bundle/EzPublishIOBundle/EventListener/StreamFileListener.php

```diff
@@ -62,7 +62,11 @@ public function onKernelRequest(RequestEvent $event)
 
         $binaryFile = $this->ioService->loadBinaryFileByUri($uri);
         if ($binaryFile instanceof MissingBinaryFile) {
-            throw new NotFoundHttpException("Could not find 'BinaryFile' with identifier '$uri'");
+            // Try again with decoded URI
+            $binaryFile = $this->ioService->loadBinaryFileByUri(urldecode($uri));
+            if ($binaryFile instanceof MissingBinaryFile) {
+                throw new NotFoundHttpException("Could not find 'BinaryFile' with identifier '$uri'");
+            }
```

Comment:
The proper solution for all these troubles is to rename ALL uploaded files to random hashes. Problem solved, and it has other benefits (unguessable). But that's a bigger change. So do we want a quick solution/workaround now or not? A real fix should come later in any case.

Reply:
@adamwojs https://github.com/ezsystems/ezpublish-kernel/pull/3126 was fixing a different issue, and with or without these changes ibx-1392 is still valid.

Reply:
Both @adamwojs and @mateuszbieniek are right, to some extent. Mateusz's PR fixed the following problematic filenames:
`some%25special+chars.jpg`. 
However, https://github.com/ezsystems/ezpublish-kernel/pull/3126 broke the following:
`test test.png`.

The current PR handles both these cases,  we can only go as temporary solution with just:
`        $binaryFile = $this->ioService->loadBinaryFileByUri(urldecode($uri));
         if ($binaryFile instanceof MissingBinaryFile) {
             throw new NotFoundHttpException("Could not find 'BinaryFile' with identifier '$uri'");
         }
`

Reply:
> Q: What will happen if my file is named `obraze%20z spacja.jpg`?

This will not work in any or before those PRs.

Reply:
> > Q: What will happen if my file is named `obraze%20z spacja.jpg`?
> 
> This will not work in any of those PRs or even before those PRs were introduced.

But do we have a solution for such case?

Reply:
Besides random hashes, I don't think so. The problem is that the browser converts the space to `%20` before a request is made so we cannot guess which `%20` is a space or which is just `%20`.

Reply:
@glye I agree but we can't apply this solution here. In most cases images are served directly from the webserver, therefore introducing hashing will either force everyone to stream files via index.php, or at least will require a change in generated URLs, which I think both are out of the question.

Reply:
(Continuing the renaming/streaming discussion elsewhere)

Reply:
> bigger

I think we should only sluggify it instead of changing it to a random name, for SEO perpective having name with keyword in it could be a good point.

Yohann